extensions/bindtextdomain/bindtextdomain.c

```diff
@@ -0,0 +1,119 @@
+/* gcc -Wall -O2 -o bindtextdomain.so -fPIC -shared bindtextdomain.c -ldl */
+
+#define _GNU_SOURCE
+
+#include <errno.h>
+#include <dirent.h>
+#include <dlfcn.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+static void die (const char *msg)
+{
+        fprintf (stderr, "can't preload: %s\n", msg);
+        exit (EXIT_FAILURE);
+}
+
+typedef char * (*BindTextDomainFunc) (const char *a,
+                                      const char *b);
+
+static BindTextDomainFunc r_bindtextdomain = NULL;
+
+char *bindtextdomain(const char *domainname, const char *dirname)
+{
+        char *snap = NULL;
+        char *snap_path = NULL;
+        char *snap_locale_path = NULL;
+        DIR *dir = NULL;
+        struct dirent *ent = NULL;
+        char *ret = NULL;
+
+        if (r_bindtextdomain == 0 && !(r_bindtextdomain =
+                                       (BindTextDomainFunc) dlsym (RTLD_NEXT, "bindtextdomain")))
+                die ("can't find symbol \"bindtextdomain\"");
+
+        if (dirname == NULL || dirname[0] != '/')
+                goto orig;
+
+        snap = getenv ("SNAP");
+
+        if (snap == NULL || strcmp (snap, "") == 0)
+                goto orig;
+
+        char * paths[] = {
+                "gnome-platform/usr/share/locale",
```

Comment:
Please add "gnome-platform/usr/share/locale-langpack" to the list of paths. @kenvandine included the gtk3 translations in the gnome-3-28-1804 snap (currently in the candidate channel) so that individual snaps don't have to do it.